src/react/databrowser/objects/details/Metadata.jsx

```diff
@@ -0,0 +1,171 @@
+    const isValidItems = useMemo(() => {
+        return !items.find(i => (i.key === '' && i.value !== '') || (i.value === '' && i.key !== ''));
+    }, [items]);
+
+    const handleKeyChange = (index: number) => (v) => {
+        const temp = [...items];
+        temp[index] = { key: v.value, value: '' };
+        setItems(temp);
+    };
+
+    const handleMetaKeyChange = (index: number) => (e: SyntheticInputEvent<HTMLInputElement>) => {
+        const temp = [...items];
+        if (!isMetaWord(temp[index].key)) {
+            return;
+        }
+        temp[index] = { ...temp[index], metaKey: e.target.value };
+        setItems(temp);
+    };
+
+    const handleValueChange = (index: number) => (e: SyntheticInputEvent<HTMLInputElement>) => {
+        const temp = [...items];
+        temp[index] = { ...temp[index], value: e.target.value };
+        setItems(temp);
+    };
+
+    const insertEntry = () => {
+        const temp = [...items];
+        temp.push(EMPTY_ITEM);
+        setItems(temp);
+    };
+
+    const deleteEntry = (index: number) => {
+        let temp = [EMPTY_ITEM];
+        if (items.length > 1) {
+            temp = [...items];
+            temp.splice(index, 1);
+        }
+        setItems(temp);
+    };
+
+    const save = () => {
+        if (!isValidItems) {
+            return;
+        }
+        const metadata = convertToAWSMetadata(items);
+        dispatch(putObjectMetadata(bucketName, prefixWithSlash, objectKey, metadata));
+    };
+
+    return (
+        <div>
+            <Header>
+                <HeaderKey> Key </HeaderKey>
+                <HeaderValue> Value </HeaderValue>
+            </Header>
+            <TableContainer>
+                <Items>
+                    {
+                        items.map((p, i) => {
+                            const isShrink = isMetaWord(p.key);
+                            return <Item isShrink={isShrink} key={i}>
+                                <Inputs>
+                                    <Select
+                                        id='mdKeyType'
+                                        name='mdKeyType'
+                                        options={options}
+                                        onChange={handleKeyChange(i)}
+                                        isDisabled={false}
+                                        value={p.key ? selectOptions.find(l => l.value === p.key): ''}
+                                    />
+                                    {
+                                        isShrink && <Char>-</Char>
+                                    }
+                                    {
+                                        isShrink && <InputExtraKey value={p.metaKey || ''} onChange={handleMetaKeyChange(i)}/>
+                                    }
+                                    <Char>:</Char>
+                                    <InputValue isShrink={isShrink} value={p.value} onChange={handleValueChange(i)}/>
```

Comment:
I think `Value` column should be always aligned with value items.

<img width="509" alt="Screenshot 2020-10-21 at 18 01 21" src="https://user-images.githubusercontent.com/3807744/96805922-968a5580-13c7-11eb-8bef-4feb2460acb4.png">